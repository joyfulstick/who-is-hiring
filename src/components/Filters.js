import React from 'react'
import FiltersStyle from '../styles/FiltersStyle'
import Input from './Input'
import Select from './Select'

const Filters = ({ handleChange, filters: { position, salary, isRemote } }) => {
  const inputs = [
    {
      name: 'salary',
      type: 'range',
      min: '0',
      max: '20',
      value: salary,
      handleChange,
    },
    { name: 'isRemote', type: 'checkbox', checked: isRemote, handleChange },
  ]
  const selects = [
    {
      name: 'position',
      value: position,
      handleChange,
      options: ['all', 'engineer', 'data scientist', 'designer', 'manager'],
    },
  ]

  return (
    <FiltersStyle>
      {selects.map(select => (
        <Select key={select.name} {...select} />
      ))}
      {inputs.map(input => (
        <Input key={input.name} {...input} />
      ))}
    </FiltersStyle>
  )
}

export default Filters
