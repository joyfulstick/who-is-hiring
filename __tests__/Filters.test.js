import { shallow } from 'enzyme'
import React from 'react'
import Filters from '../src/components/Filters'

describe('<Filters />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Filters
        handleChange={jest.fn()}
        filters={{ position: 'all', salary: 5, isRemote: false }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
