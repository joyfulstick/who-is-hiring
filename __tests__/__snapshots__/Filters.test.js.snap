// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> renders correctly 1`] = `
<styled.div>
  <Select
    handleChange={[MockFunction]}
    key="position"
    name="position"
    options={
      Array [
        "all",
        "engineer",
        "data scientist",
        "designer",
        "manager",
      ]
    }
    value="all"
  />
  <Input
    handleChange={[MockFunction]}
    key="salary"
    max="20"
    min="0"
    name="salary"
    type="range"
    value={5}
  />
  <Input
    checked={false}
    handleChange={[MockFunction]}
    key="isRemote"
    name="isRemote"
    type="checkbox"
  />
</styled.div>
`;
